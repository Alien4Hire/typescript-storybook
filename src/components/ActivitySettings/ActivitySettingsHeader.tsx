import { Box, Button, Typography } from '@mui/material';

type ActivitySettingsHeaderT = {
  onClickCancel: () => void;
  onClickSave: () => void;
  testId?: string;
};

export function ActivitySettingsHeader({
  onClickCancel,
  onClickSave,
  testId = '',
}: ActivitySettingsHeaderT) {
  return (
    <Box
      sx={{
        display: 'flex',
        justifyContent: 'space-between',
        alignItems: 'center',
        paddingBottom: '10px',
        paddingTop: '10px',
      }}
      data-testid={testId}
    >
      <Typography variant="h2" sx={{ fontWeight: 700 }}>
        Activity Settings
      </Typography>
      <Box sx={{ display: 'flex' }}>
        <Button
          data-testid="cancel-activity-settings-header-btn"
          variant="outlined"
          onClick={onClickCancel}
          sx={{ textTransform: 'none' }}
        >
          Cancel
        </Button>
        <Button
          data-testid="save-activity-settings-header-btn"
          variant="contained"
          onClick={onClickSave}
          sx={{ marginLeft: 2, textTransform: 'none' }}
        >
          Save
        </Button>
      </Box>
    </Box>
  );
}
