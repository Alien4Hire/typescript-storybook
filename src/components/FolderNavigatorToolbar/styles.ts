import { Box, Button } from '@mui/material';

import { styled } from '@mui/material/styles';

export const Divider = styled('div')(({ theme }) => ({
  borderRight: `1px solid ${theme.palette.gray.medium}`,
  height: '34px',
  margin: 'auto',
}));

export const Label = styled('div')(({ theme }) => ({
  fontWeight: '600',
  fontSize: '15px',
  letterSpacing: '0.4px',
  color: theme.palette.gray.veryDark,
  fontFamily: "'Source Sans Pro', sans-serif",
}));

export const Wrapper = styled('div')(({ theme }) => ({
  position: 'relative',
  background: theme.palette.gray.veryLight,
  paddingLeft: '56px',
}));

export const FolderForm = styled('div')`
  display: flex;
  align-items: center;
`;

export const FolderAdd = styled(Box, {
  shouldForwardProp: (prop: string) => prop !== 'active',
})(({ active, theme }) => ({
  position: 'absolute',
  top: 0,
  bottom: 0,
  left: 0,
  right: 0,
  width: 'auto',
  padding: '2px',
  backgroundColor: theme.palette?.gray.veryLight,
  display: 'none',
  ...(active && {
    display: 'flex',
    alignItems: 'center',
  }),
}));

export const CsvModalBackdrop = styled(Box)({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  backgroundColor: '#f1f1f1',
  borderRadius: '16px',
  padding: '32px 32px 70px',
});

export const UploadListItem = styled(Box)(({ theme }) => ({
  display: 'flex',
  alignItems: 'center',
  padding: '16px 0px',
  borderBottom: `1px solid ${theme.palette.gray.light}`,
}));

export const UploadButton = styled(Button)({
  textTransform: 'none',
  width: 'fit-content',
  marginTop: '16px',
});

export const DownloadTemplateWrapper = styled(Box)({
  display: 'flex',
  alignItems: 'center',
  marginTop: '24px',
  cursor: 'pointer',
});
