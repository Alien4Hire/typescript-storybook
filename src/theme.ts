import { createTheme } from '@mui/material/styles';
import mlColors from './util/ml-cdl-colors';
import {
  computerSmallMonitorMax,
  mobileScreenMax,
  mobileScreenMin,
  tabletScreenMax,
} from './util/breakpoints';

declare module '@mui/material/Typography' {
  interface TypographyPropsVariantOverrides {
    title: true;
    textLarge: true;
    textBasic: true;
    textSmall: true;
    textSub: true;
    buttonMedium: true;
    buttonSmall: true;
    buttonLarge: true;
  }
}

//UX variants definitions can be found at https://www.figma.com/file/bEonWYX4lesmqlFY6MqWCt/Course-Assembly-MVP?node-id=5540%3A249648
//Custom created variant names for typographies

declare module '@mui/material/styles/createTypography' {
  interface Typography {
    formTitle?: React.CSSProperties;
    formSubtitle?: React.CSSProperties;
    appTitle?: React.CSSProperties;
    title?: React.CSSProperties;
    textLarge?: React.CSSProperties;
    textBasic?: React.CSSProperties;
    textSmall?: React.CSSProperties;
    textSub?: React.CSSProperties;
    buttonMedium?: React.CSSProperties;
    buttonSmall?: React.CSSProperties;
    buttonLarge?: React.CSSProperties;
  }

  // allow configuration using `createMuiTheme`
  interface TypographyOptions {
    title?: React.CSSProperties;
    textLarge?: React.CSSProperties;
    textBasic?: React.CSSProperties;
    textSmall?: React.CSSProperties;
    textSub?: React.CSSProperties;
    buttonMedium?: React.CSSProperties;
    buttonSmall?: React.CSSProperties;
    buttonLarge?: React.CSSProperties;
  }
}

declare module '@mui/material/styles' {
  interface Theme {}
  interface ThemeOptions {}
  interface Palette {
    specials: Palette['primary'];
    aqua: Palette['primary'];
    black: Palette['primary'];
    blue: Palette['primary'];
    gray: Palette['primary'];
    green: Palette['primary'];
    magenta: Palette['primary'];
    orange: Palette['primary'];
    purple: Palette['primary'];
    red: Palette['primary'];
    teal: Palette['primary'];
    white: Palette['primary'];
    yellow: Palette['primary'];
  }
  interface PaletteOptions {
    specials: PaletteOptions['primary'];
    aqua: PaletteOptions['primary'];
    black: PaletteOptions['primary'];
    blue: PaletteOptions['primary'];
    gray: PaletteOptions['primary'];
    green: PaletteOptions['primary'];
    magenta: PaletteOptions['primary'];
    orange: PaletteOptions['primary'];
    purple: PaletteOptions['primary'];
    red: PaletteOptions['primary'];
    teal: PaletteOptions['primary'];
    white: PaletteOptions['primary'];
    yellow: PaletteOptions['primary'];
  }

  interface PaletteColor {
    macmillanRed?: string;
    offWhite?: string;
    veryLight?: string;
    medium?: string;
    veryDark?: string;
    fullPageDialogShadow?: string;
  }

  interface SimplePaletteColorOptions {
    macmillanRed?: string;
    offWhite?: string;
    veryLight?: string;
    medium?: string;
    veryDark?: string;
    fullPageDialogShadow?: string;
  }
}

export const theme = createTheme({
  palette: {
    background: {
      default: mlColors.off_white,
      paper: mlColors.pure_white,
    },
    primary: {
      main: mlColors.aqua,
    },
    secondary: {
      main: mlColors.aqua_dark,
    },
    specials: {
      main: mlColors.macmillan_red,
      macmillanRed: mlColors.macmillan_red,
      fullPageDialogShadow: 'rgba(0, 0, 0, 0.5)',
    },
    aqua: {
      main: mlColors.aqua,
      veryLight: mlColors.aqua_very_light,
      light: mlColors.aqua_light,
      medium: mlColors.aqua_medium,
      dark: mlColors.aqua_dark,
      veryDark: mlColors.dark_navy,
    },
    black: {
      main: mlColors.black,
    },
    blue: {
      main: mlColors.blue,
      veryLight: mlColors.blue_very_light,
      light: mlColors.blue_light,
      medium: mlColors.blue_medium,
      dark: mlColors.blue_dark,
    },
    gray: {
      main: mlColors.gray,
      veryLight: mlColors.gray_very_light,
      light: mlColors.gray_light,
      medium: mlColors.gray_medium,
      dark: mlColors.gray_dark,
      veryDark: mlColors.gray_very_dark,
    },
    green: {
      main: mlColors.green,
      veryLight: mlColors.green_very_light,
      light: mlColors.green_light,
      medium: mlColors.green_medium,
      dark: mlColors.green_dark,
    },
    magenta: {
      main: mlColors.magenta,
      veryLight: mlColors.magenta_very_light,
      light: mlColors.magenta_light,
      medium: mlColors.magenta_medium,
      dark: mlColors.magenta_dark,
    },
    orange: {
      main: mlColors.orange,
      veryLight: mlColors.orange_very_light,
      light: mlColors.orange_light,
      medium: mlColors.orange_medium,
      dark: mlColors.orange_dark,
    },
    purple: {
      main: mlColors.purple,
      veryLight: mlColors.purple_very_light,
      light: mlColors.purple_light,
      medium: mlColors.purple_medium,
      dark: mlColors.purple_dark,
    },
    red: {
      main: mlColors.red,
      veryLight: mlColors.red_very_light,
      light: mlColors.red_light,
      medium: mlColors.red_medium,
      dark: mlColors.red_dark,
    },
    teal: {
      main: mlColors.teal,
      veryLight: mlColors.teal_very_light,
      light: mlColors.teal_light,
      medium: mlColors.teal_medium,
      dark: mlColors.teal_dark,
      veryDark: mlColors.teal_very_dark,
    },
    white: {
      main: mlColors.pure_white,
      offWhite: mlColors.off_white,
    },
    yellow: {
      main: mlColors.yellow,
      veryLight: mlColors.yellow_very_light,
      light: mlColors.yellow_light,
      medium: mlColors.yellow_medium,
      dark: mlColors.yellow_dark,
    },
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: mobileScreenMin,
      md: mobileScreenMax,
      lg: tabletScreenMax,
      xl: computerSmallMonitorMax,
    },
  },
  spacing: 8,
  typography: {
    fontSize: 14,
    fontFamily: [
      '-apple-system',
      'BlinkMacSystemFont',
      '"Segoe UI"',
      'Roboto',
      'Oxygen',
      'Ubuntu',
      'Cantarell',
      '"Fira Sans"',
      '"Droid Sans"',
      '"Helvetica Neue"',
      'Source Sans Pro',
      'sans-serif',
    ].join(','),
    allVariants: {
      fontWeight: 'normal',
    },
    button: {
      fontSize: 14,
      textTransform: 'uppercase',
    },
    h1: {
      color: mlColors.black,
      fontSize: 32,
      fontFamily: 'Source Sans Pro',
      fontWeight: 600,
    },
    h2: {
      color: mlColors.black,
      fontSize: 24,
      fontFamily: 'Source Sans Pro',
      fontWeight: 600,
    },
    h3: {
      color: mlColors.black,
      fontSize: 18,
      fontFamily: 'Source Sans Pro',
      fontWeight: 600,
    },
    h4: {
      color: mlColors.black,
      fontSize: 16,
      fontFamily: 'Source Sans Pro',
      fontWeight: 600,
    },
    title: {
      color: mlColors.black,
      fontSize: 28,
      fontFamily: 'Source Sans Pro',
      fontWeight: 200,
      fontStyle: 'normal',
    },
    textLarge: {
      color: mlColors.black,
      fontSize: 18,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    textBasic: {
      color: mlColors.black,
      fontSize: 16,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    textSmall: {
      color: mlColors.black,
      fontSize: 14,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    textSub: {
      color: mlColors.black,
      fontSize: 13,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    buttonLarge: {
      color: mlColors.black,
      fontSize: 15,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    buttonMedium: {
      color: mlColors.black,
      fontSize: 18,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    buttonSmall: {
      color: mlColors.black,
      fontSize: 12,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
  },
  components: {
    MuiAlert: {
      styleOverrides: {
        filledError: {
          background: mlColors.red_very_light,
          color: mlColors.black,
          border: `1px solid ${mlColors.red}`,
        },
        filledInfo: {
          background: mlColors.blue_very_light,
          color: mlColors.black,
          border: `1px solid ${mlColors.blue}`,
        },
        filledSuccess: {
          background: mlColors.green_very_light,
          color: mlColors.black,
          border: `1px solid ${mlColors.green}`,
        },
        filledWarning: {
          background: mlColors.yellow_very_light,
          color: mlColors.black,
          border: `1px solid ${mlColors.yellow}`,
        },
      },
    },
    // MuiButton: {
    //   styleOverrides: {
    //     root: {
    //       fontSize: '1rem',
    //       textTransform: 'none',
    //       height: '35px',
    //       padding: '0 15px',
    //       margin: '5px',
    //       '&:focus': {
    //         boxShadow: '0 0 6px rgba(0, 117, 142, 1)',
    //         border: '1px solid #FFFFFF',
    //       },
    //       '&:disabled': {
    //         background: '#c8c8c8',
    //         color: '#f3f3f3',
    //       },
    //     },
    //     textSecondary: {
    //       fontSize: '0.9375rem',
    //       height: '34px',
    //       lineHeight: '33px',
    //       padding: '0 15px',
    //       borderRadius: '3px',
    //       '-webkit-border-radius': '3px',
    //       '-ms-border-radius': '3px',
    //       '-moz-border-radius': '3px',
    //       textDecoration: 'none',
    //       color: '#00758e',
    //       cursor: 'pointer',
    //       background: 'transparent',
    //       border: '1px solid #00758e',
    //       overflow: 'hidden',
    //     },
    //     containedPrimary: {
    //       '&:focus': {
    //         boxShadow: '0 0 6px rgba(0, 117, 142, 1)',
    //         border: '1px solid #FFFFFF',
    //       },
    //       '&:disabled': {
    //         background: '#c8c8c8',
    //         color: '#f3f3f3',
    //       },
    //     },
    //     outlinedSecondary: {
    //       '&:focus': {
    //         boxShadow: '0 0 6px rgba(0, 117, 142, 1)',
    //         border: '1px solid #00758e',
    //       },
    //       '&:disabled': {
    //         color: '#989898',
    //         fill: '#c8c8c8',
    //         backgroundColor: '#f3f3f3',
    //       },
    //     },
    //   },
    // },
    MuiButtonBase: {
      defaultProps: {
        disableRipple: true, // No more ripple, on the whole application!
        focusRipple: false,
      },
    },
  },
});
