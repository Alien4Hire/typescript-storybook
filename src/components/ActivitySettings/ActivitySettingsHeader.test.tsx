import { render, screen } from '@testing-library/react';
import { ThemeProvider, StyledEngineProvider } from '@mui/material/styles';

import { ActivitySettingsHeader } from './ActivitySettingsHeader';

import { theme } from '../../theme';
import '../../setupTests';

const baseProps = {
  onClickCancel: () => {},
  onClickSave: () => {},
};

const renderComponent = (customProps?: { testId?: string }) => {
  const props = { ...baseProps, ...customProps };

  return render(
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={theme}>
        <ActivitySettingsHeader {...props} />
      </ThemeProvider>
    </StyledEngineProvider>
  );
};

describe('<ActivitySettingsHeader />', () => {
  it('renders without crashing', () => {
    const view = renderComponent();
    expect(view).toMatchSnapshot();
  });

  it('should have a title', () => {
    renderComponent();
    expect(screen.getByText('Activity Settings')).toBeInTheDocument();
  });

  it('should have a cancel button', () => {
    renderComponent();
    expect(screen.getByText('Cancel')).toBeInTheDocument();
  });

  it('should have a save button', () => {
    renderComponent();
    expect(screen.getByText('Save')).toBeInTheDocument();
  });

  it('should have a data-test-id', () => {
    const testId = 'test-activity-settings-header';
    renderComponent({ testId });
    const element = screen.getByTestId(testId);
    expect(element).toBeInTheDocument();
  });
});
