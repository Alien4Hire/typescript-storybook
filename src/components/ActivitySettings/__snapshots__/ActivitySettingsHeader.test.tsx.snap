// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActivitySettingsHeader /> renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c3::-moz-focus-inner {
  border-style: none;
}

.c3.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c1 {
  margin: 0;
  color: #080808;
  font-size: 24px;
  font-family: Source Sans Pro;
  font-weight: 600;
  line-height: 1.2;
  font-weight: 700;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-bottom: 10px;
  padding-top: 10px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  font-size: 14px;
  text-transform: uppercase;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  line-height: 1.75;
  min-width: 64px;
  padding: 5px 15px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,border-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,color 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,border-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,color 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  border: 1px solid rgba(0,117,142,0.5);
  color: #00758e;
  text-transform: none;
}

.c4:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0,117,142,0.04);
  border: 1px solid #00758e;
}

.c4.Mui-disabled {
  color: rgba(0,0,0,0.26);
  border: 1px solid rgba(0,0,0,0.12);
}

.c5 {
  font-size: 14px;
  text-transform: uppercase;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  line-height: 1.75;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,border-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,color 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,border-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,color 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  color: #fff;
  background-color: #00758e;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  margin-left: 16px;
  text-transform: none;
}

.c5:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgb(0,81,99);
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

.c5:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.c5.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.c5.Mui-disabled {
  color: rgba(0,0,0,0.26);
  box-shadow: none;
  background-color: rgba(0,0,0,0.12);
}

@media print {
  .c3 {
    color-adjust: exact;
  }
}

@media (hover:none) {
  .c4:hover {
    background-color: transparent;
  }
}

@media (hover:none) {
  .c5:hover {
    background-color: #00758e;
  }
}

<body>
    <div>
      <div
        class="c0 MuiBox-root"
        data-testid=""
      >
        <h2
          class="c1 MuiTypography-root MuiTypography-h2"
        >
          Activity Settings
        </h2>
        <div
          class="c2 MuiBox-root"
        >
          <button
            class="c3 MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root c4"
            data-testid="cancel-activity-settings-header-btn"
            tabindex="0"
            type="button"
          >
            Cancel
          </button>
          <button
            class="c3 MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root c5"
            data-testid="save-activity-settings-header-btn"
            tabindex="0"
            type="button"
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": .c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c3::-moz-focus-inner {
  border-style: none;
}

.c3.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c1 {
  margin: 0;
  color: #080808;
  font-size: 24px;
  font-family: Source Sans Pro;
  font-weight: 600;
  line-height: 1.2;
  font-weight: 700;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-bottom: 10px;
  padding-top: 10px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  font-size: 14px;
  text-transform: uppercase;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  line-height: 1.75;
  min-width: 64px;
  padding: 5px 15px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,border-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,color 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,border-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,color 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  border: 1px solid rgba(0,117,142,0.5);
  color: #00758e;
  text-transform: none;
}

.c4:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0,117,142,0.04);
  border: 1px solid #00758e;
}

.c4.Mui-disabled {
  color: rgba(0,0,0,0.26);
  border: 1px solid rgba(0,0,0,0.12);
}

.c5 {
  font-size: 14px;
  text-transform: uppercase;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  line-height: 1.75;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,border-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,color 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,border-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,color 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  color: #fff;
  background-color: #00758e;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  margin-left: 16px;
  text-transform: none;
}

.c5:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgb(0,81,99);
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

.c5:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.c5.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.c5.Mui-disabled {
  color: rgba(0,0,0,0.26);
  box-shadow: none;
  background-color: rgba(0,0,0,0.12);
}

@media print {
  .c3 {
    color-adjust: exact;
  }
}

@media (hover:none) {
  .c4:hover {
    background-color: transparent;
  }
}

@media (hover:none) {
  .c5:hover {
    background-color: #00758e;
  }
}

<div>
    <div
      class="c0 MuiBox-root"
      data-testid=""
    >
      <h2
        class="c1 MuiTypography-root MuiTypography-h2"
      >
        Activity Settings
      </h2>
      <div
        class="c2 MuiBox-root"
      >
        <button
          class="c3 MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root c4"
          data-testid="cancel-activity-settings-header-btn"
          tabindex="0"
          type="button"
        >
          Cancel
        </button>
        <button
          class="c3 MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root c5"
          data-testid="save-activity-settings-header-btn"
          tabindex="0"
          type="button"
        >
          Save
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
